docs/design/solutions/pce-weights-strategies/pce-weights-strategies.md

```diff
@@ -0,0 +1,19 @@
+# Refactor PCE with weights computation strategies
+
+## Goals
+For now PCE uses only cost field as graph edges weights
+
+While path computing logic going to be more complicated - we need to take several factors\costs to compute final edge\node weight, and keep this things maintainable.
+Also built AvailableNetwork should be logged, to make strange computed paths investigation possible.
+
+## Proposed changes
+- Introduce PCE data model instead of reusing DAO: for storing graph weighs by features and more convenient logging
+- Separate AvailableNetwork creation: it isn't responsibility of PathComputer
+- Introduce weights computing strategies: like weight getters above data model, defines what weigh features will be used in path computation and in that proportions
+
+### Weight strategy example
+Suppose that we have `sw1 -> isl -> sw2` graph edge in diversity scenario, then weight strategy may looks like:
+`sw1 diversity cost + isl cost + isl diversity cost + sw2 diversity cost`.
+
```

Comment:
We doesn't needed entire set of paths, it's only a new flow feature for now, and it will be diversed only with one\several other flows.
Fore this case we are going to change "diversity flag" to some configurable big constant, gets it as a part of final edge weight with the proper strategy, and keep current algo working. If there is a path around used segment - it will be returned, if not - solution allows partial diversity, for cases like bottlenecks between graph parts.